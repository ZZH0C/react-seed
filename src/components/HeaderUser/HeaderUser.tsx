import React, { useContext } from 'react';
import { UserContext } from '../../pages/HomePage/HomePage';

interface HeaderUserProps extends React.HTMLAttributes<HTMLElement> {
  src?: string;
}

export const HeaderUser: React.FC<HeaderUserProps> = ({ src, children }) => {
  const userData = useContext(UserContext);
  const profile = {
    name: '',
    iconSrc: src,
  };
  if (userData && 'profileObj' in userData) {
    profile.name = userData.profileObj.name;
    profile.iconSrc = userData.profileObj.imageUrl;
  }
  return (
    <div className="header-comp pull-right">
      {children}
      <a href="/" className="profile">
        <span>{profile.name}</span>
        <img alt={'userPick'} src={profile.iconSrc} />
      </a>
      <a href="/" className="btn btn-xs btn-header">
        <i className="headerIcon icon-search" />
      </a>
    </div>
  );
};
