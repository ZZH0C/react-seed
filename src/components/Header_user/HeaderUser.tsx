import React from 'react';
import { HeaderUserIcons } from '../Header_user_icons/HeaderUserIcons';
type HeaderUserProps = React.HTMLAttributes<HTMLElement>;

export const HeaderUser: React.FC<HeaderUserProps> = () => {
  return (
    <div className="header-comp pull-right">
      <HeaderUserIcons style={'headerIcon icon-bell'} />
      <HeaderUserIcons style={'headerIcon icon-mail'} />
      <a href="./index.html" className="profile">
        <span>User name</span>
        <img alt={'Search icon'} src="images/profilePhoto.png" />
      </a>
      <a href="./index.html" className="btn btn-xs btn-header">
        <i className="headerIcon icon-search" />
      </a>
    </div>
  );
};
