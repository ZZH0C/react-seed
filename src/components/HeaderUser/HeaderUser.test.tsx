import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });
import { HeaderUser } from './HeaderUser';

describe('components/ButtonLink', () => {
  const mockProps = {
    href: '/',
    children: 'some text',
  };

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should render and match snapshot', () => {
    expect(
      shallow(
        <HeaderUser src={mockProps.href}>{mockProps.children}</HeaderUser>,
      ),
    ).toMatchSnapshot();
  });
  //TODO: add useContext test
});
