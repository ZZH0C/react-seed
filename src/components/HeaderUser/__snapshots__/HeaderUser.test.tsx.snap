// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/ButtonLink should render and match snapshot 1`] = `
<div
  className="header-comp pull-right"
>
  some text
  <a
    className="profile"
    href="/"
  >
    <span />
    <img
      alt="userPick"
      src="/"
    />
  </a>
  <a
    className="btn btn-xs btn-header"
    href="/"
  >
    <i
      className="headerIcon icon-search"
    />
  </a>
</div>
`;
